### PR TITLE
Add Jest testing skeleton and one snapshot test

### PR DESCRIPTION
**Pull request by jasonbcox**
This introduces a framework for writing various types of tests, as well as a very bare-bones test so that we know everything works.

Most of this is unit test scaffolding pulled from https://www.gatsbyjs.com/docs/unit-testing/

The tweaks to PrimaryButton are inspired by https://www.gatsbyjs.com/docs/testing-components-with-graphql/

#### Test Plan
```
npm install
npm run test
```
And for sanity:
```
npm run develop
```
Navigate to localhost:8000

### CODE REVIEW

File: v2/jest/preprocess.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const babelOptions = {
+  presets: ["babel-preset-gatsby"],
+}
+
+module.exports = require("babel-jest").createTransformer(babelOptions)
```
Comment (deadalnix) [inline v2/jest/preprocess.js:5]: That would be preferable if this could use the .babelrc provided. This is not going to handle fbt properly for instance.

**Review by deadalnix — CHANGES_REQUESTED** [review]
That seems like a lot of crap to just test one button XD

Is this really how people do this in the JS ecosystem?

File: v2/jest/preprocess.js (hunk 1 above)
Comment (jasonbcox) [reply v2/jest/preprocess.js]: Apparently it uses .babelrc out of the box. Not sure why this note is hidden away in their docs, but everything appears to work just removing this file all together and getting rid of the explicit `transform:` config line.

**Comment by jasonbcox** [comment]
> That seems like a lot of crap to just test one button XD
> 
> Is this really how people do this in the JS ecosystem?

Funny you say this, because this might be the best state I've seen the JS ecosystem in over the last 10 years LOL

**Pull request merged**